### ISSUE TITLE
it simply doesn't work

### ISSUE THREAD

**Issue by BadIDMan** [Magisk-Modules-Alt-Repo/ezme-nodebug#2]
Im running LineageOS Android 9 with Magisk 26.1. Installed "Hide UserDebug, Test-Keys and LineageOS" v1 by ez-me but Momo 4.4.1 says "Debugging mode is enabled"
When I disable Android debugging" in Developer options then Momo is not detecting debugging mode is enabled because it is not enabled.
So "Hide UserDebug, Test-Keys and LineageOS" simply doesn't work.

**Comment by ez-me** [comment]
UserDebug is how the ROM is compiled, not if you have USB Debugging enabled or not.

**Comment by bengalih** [comment]
> UserDebug is how the ROM is compiled, not if you have USB Debugging enabled or not.

I don't think you are correct.  Momo will state "Debugging mode is enabled" when USB debugging is enabled and it will not report it when USB debugging is turned off.

I discovered your module when searching for HuskyDG's old hide_userdebug module.  I had that module running on my phone before I reset it (same ROM and setup) and with that module Momo was successfully fooled into not showing that warning when USB debugging was enabled.

Unfortunately, Husky's new module does not seem to work to do this, and neither does yours.  Husky's seem to issue a command to fix it:
https://github.com/Magisk-Modules-Alt-Repo/sensitive_props/blob/05ab33d0de6ebea2161e905dd2cad27fcd5d3b44/oem.rc#L7

`resetprop -n init.svc.adbd stopped`

I don't see any similar code in yours.

I'm trying all over the place to find an old copy of his .zip to validate, but he seems to have removed all traces.

UPDATE:

Despite that code being in HuskyDG's updated module, it does not appear to work either.

Using Momo 4.3.1 neither your module or sensitive_props hides the USB debug message.
However creating a script in /data/adb/service.d that includes the above line successfully hides the message.

However, in Momo 4.4.1 not even this bypasses it.

Apparently Momo has implemented some additional check that nothing (so far) can trick.

That being said, this module doesn't do it and neither does sensitive_props in any case.

**Comment by etillxd** [comment]
> I don't think you are correct.  Momo will state "Debugging mode is enabled" when USB debugging is enabled and it will not report it when USB debugging is turned off.
> 

This module has nothing to do with hiding USB Debugging, hence why momo correctly detects it if it's enabled.

The UserDebug this module tries to hide, is the Buildtype of the ROM, one of 3 to be exact (user, userdebug and eng), which is defined before compiling and then decides how the ROM is compiled. See https://source.android.com/docs/setup/build/building?hl=en under Buildtype.

**Comment by bengalih** [comment]
> > I don't think you are correct.  Momo will state "Debugging mode is enabled" when USB debugging is enabled and it will not report it when USB debugging is turned off.
> 
> This module has nothing to do with hiding USB Debugging, hence why momo correctly detects it if it's enabled.
> 
> The UserDebug this module tries to hide, is the Buildtype of the ROM, one of 3 to be exact (user, userdebug and eng), which is defined before compiling and then decides how the ROM is compiled. See https://source.android.com/docs/setup/build/building?hl=en under Buildtype.

Thanks for the clarification, I was directed here through an outside source initially and thus did not see the full description.  I was confused due to the author's call out to the original Husky module which included the resetprop command I mentioned above whose purpose was to hide USB debugging (at least the purpose of that command).
